[Bug]: panic: re submit special order would cross - after zero volume changes

### Problem encountered

Overnight market-sim fuzzing occasionally hitting panic in runs involving AMMs after zero volume changes.

https://jenkins.vega.rocks/job/common/job/vega-market-sim-fuzzing-develop/33/artifact/network_home/vega-sim-irphb8z/
```
panic: re submit special order would cross

goroutine 26424 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2?, 0x2?, {0x0?, 0x0?, 0xc008ad9fa0?})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00628e680, {0xc005f79880, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0x70507845fe2498f?, {0x40cccd3?, 0x0?}, {0xc005f79880, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/logger.go:258 +0x51
code.vegaprotocol.io/vega/core/matching.(*OrderBook).ReSubmitSpecialOrders(0xc0009aca00, 0xc001990270)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/matching/orderbook.go:937 +0x9b6
code.vegaprotocol.io/vega/core/execution/future.(*Market).reSubmitPeggedOrders(0xc00076fc00, {0x4892038, 0xc008b66ff0}, {0xc0007c6800, 0x48, 0x20?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/special_orders.go:112 +0xf3
code.vegaprotocol.io/vega/core/execution/future.(*Market).repriceAllSpecialOrders(0xc00076fc00, {0x4892038, 0xc008b66ff0}, 0xb0?, {0xc005f44d20?, 0x4892038?, 0xc008b66ff0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/special_orders.go:151 +0x117
code.vegaprotocol.io/vega/core/execution/future.(*Market).checkForReferenceMoves(0xc00076fc00, {0x4892038, 0xc008b66ff0}, {0xc005f44d20, 0x3, 0x3}, 0x0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/reference_price_moves.go:54 +0x2b1
code.vegaprotocol.io/vega/core/execution/future.(*Market).handleConfirmation(0xc00076fc00, {0x4892038, 0xc008b66ff0}, 0xc005f4f740, 0x0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2951 +0x1689
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitValidatedOrder(0xc00076fc00, {0x4892038?, 0xc008b66ff0}, 0xc006e3d1e0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2674 +0x2817
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitOrder(0xc00076fc00, {0x4892038, 0xc008b66ff0}, 0xc006e3d1e0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2448 +0x25b
code.vegaprotocol.io/vega/core/execution/future.(*Market).SubmitOrderWithIDGeneratorAndOrderID(0xc00076fc00, {0x4892038?, 0xc008b66ff0}, 0xc00539d250?, {0xc004dafec0, 0x40}, {0x4860c60?, 0xc005f44870?}, {0xc005f35f80, 0x40}, ...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2407 +0x646
code.vegaprotocol.io/vega/core/execution.(*Engine).SubmitOrder(0xc000a69880, {0x4892038, 0xc008b66ff0}, 0xc00600ad80, {0xc004dafec0, 0x40}, {0x4860c60, 0xc005f44870}, {0xc005f35f80, 0x40})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/engine.go:976 +0x4af
code.vegaprotocol.io/vega/core/processor.(*BMIProcessor).ProcessBatch(0xc00539d958, {0x4892038, 0xc008b66ff0}, 0xc0059a03c0, {0xc004dafec0, 0x40}, {0xc005f35ec0, 0x40}, {0x48e7548, 0xc000658d80})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/batch_market_instructions_processor.go:241 +0x1483
code.vegaprotocol.io/vega/core/processor.(*App).DeliverBatchMarketInstructions(0xc000a69c00, {0x4892038, 0xc008b66ff0}, {0x48bf850, 0xc004dc6040}, {0xc005f35ec0, 0x40})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:1875 +0x151
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func50({0x4892038, 0xc008b66ff0}, {0x48bf850, 0xc004dc6040})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:595 +0x5e
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).CheckBatchMarketInstructionsW.func51({0x4892038, 0xc008b66ff0}, {0x48bf850, 0xc004dc6040})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:628 +0x6e
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func52({0x4892038, 0xc008b66ff0}, {0x48bf850?, 0xc004dc6040?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:658 +0x5d
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc000658a80, {0xb8?, 0x3cd6820?}, 0xc004d90d80)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/abci/abci.go:179 +0x622
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc004e21a40?, {0x4892000?, 0x6e5f000?}, 0xc001c004e0?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x25
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc0005d6a20)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:212 +0x556
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc0005d6a20, 0x6c088e200)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:289 +0x1ce
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:132 +0x25
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 26422
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:131 +0x22f
```

### Observed behaviour

-

### Expected behaviour

-

### Steps to reproduce

```bash
replay
```


### Software version

62066e8aa2f42d3579595317ad38d924efcea1c1

### Failing test

-

### Jenkins run

-

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "0000000000000000000000000000000000000000000000000000000000000000": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "quantum": "1",
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "blockchains.evmBridgeConfigs": "{\"configs\":[{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }]}",
      "blockchains.ethereumRpcAndEvmCompatDataSourcesConfig": "{\"configs\": []}",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeRebateProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.minClose": "1s",
      "governance.proposal.VolumeRebateProgram.minEnact": "1s",
      "governance.proposal.VolumeRebateProgram.minProposerBalance": "1",
      "governance.proposal.VolumeRebateProgram.minVoterBalance": "1",
      "governance.proposal.VolumeRebateProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeRebateProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "limits.markets.ammPoolEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.fee.factors.treasuryFee": "0.0001",
      "market.fee.factors.buybackFee": "0.0001",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "0000000000000000000000000000000000000000000000000000000000000000",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "spam.order.minimumMarginQuantumMultiple": "0",
      "spam.order.minimumHoldingQuantumMultiple": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "0",
      "transfer.fee.maxQuantumAmount": "100",
      "transfer.feeDiscountMinimumTrackedAmount": "0.001",
      "transfer.feeDiscountDecayFraction": "0.5",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1",
      "market.aggressiveOrderBlockDelay": "1",
      "market.amm.minCommitmentQuantum": "1",
      "market.liquidity.maxAmmCalculationLevels": "100"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: re submit special order would cross

goroutine 26424 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x2?, 0x2?, {0x0?, 0x0?, 0xc008ad9fa0?})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00628e680, {0xc005f79880, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0x70507845fe2498f?, {0x40cccd3?, 0x0?}, {0xc005f79880, 0x2, 0x2})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/logger.go:258 +0x51
code.vegaprotocol.io/vega/core/matching.(*OrderBook).ReSubmitSpecialOrders(0xc0009aca00, 0xc001990270)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/matching/orderbook.go:937 +0x9b6
code.vegaprotocol.io/vega/core/execution/future.(*Market).reSubmitPeggedOrders(0xc00076fc00, {0x4892038, 0xc008b66ff0}, {0xc0007c6800, 0x48, 0x20?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/special_orders.go:112 +0xf3
code.vegaprotocol.io/vega/core/execution/future.(*Market).repriceAllSpecialOrders(0xc00076fc00, {0x4892038, 0xc008b66ff0}, 0xb0?, {0xc005f44d20?, 0x4892038?, 0xc008b66ff0?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/special_orders.go:151 +0x117
code.vegaprotocol.io/vega/core/execution/future.(*Market).checkForReferenceMoves(0xc00076fc00, {0x4892038, 0xc008b66ff0}, {0xc005f44d20, 0x3, 0x3}, 0x0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/reference_price_moves.go:54 +0x2b1
code.vegaprotocol.io/vega/core/execution/future.(*Market).handleConfirmation(0xc00076fc00, {0x4892038, 0xc008b66ff0}, 0xc005f4f740, 0x0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2951 +0x1689
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitValidatedOrder(0xc00076fc00, {0x4892038?, 0xc008b66ff0}, 0xc006e3d1e0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2674 +0x2817
code.vegaprotocol.io/vega/core/execution/future.(*Market).submitOrder(0xc00076fc00, {0x4892038, 0xc008b66ff0}, 0xc006e3d1e0)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2448 +0x25b
code.vegaprotocol.io/vega/core/execution/future.(*Market).SubmitOrderWithIDGeneratorAndOrderID(0xc00076fc00, {0x4892038?, 0xc008b66ff0}, 0xc00539d250?, {0xc004dafec0, 0x40}, {0x4860c60?, 0xc005f44870?}, {0xc005f35f80, 0x40}, ...)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/future/market.go:2407 +0x646
code.vegaprotocol.io/vega/core/execution.(*Engine).SubmitOrder(0xc000a69880, {0x4892038, 0xc008b66ff0}, 0xc00600ad80, {0xc004dafec0, 0x40}, {0x4860c60, 0xc005f44870}, {0xc005f35f80, 0x40})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/execution/engine.go:976 +0x4af
code.vegaprotocol.io/vega/core/processor.(*BMIProcessor).ProcessBatch(0xc00539d958, {0x4892038, 0xc008b66ff0}, 0xc0059a03c0, {0xc004dafec0, 0x40}, {0xc005f35ec0, 0x40}, {0x48e7548, 0xc000658d80})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/batch_market_instructions_processor.go:241 +0x1483
code.vegaprotocol.io/vega/core/processor.(*App).DeliverBatchMarketInstructions(0xc000a69c00, {0x4892038, 0xc008b66ff0}, {0x48bf850, 0xc004dc6040}, {0xc005f35ec0, 0x40})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:1875 +0x151
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func50({0x4892038, 0xc008b66ff0}, {0x48bf850, 0xc004dc6040})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:595 +0x5e
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).CheckBatchMarketInstructionsW.func51({0x4892038, 0xc008b66ff0}, {0x48bf850, 0xc004dc6040})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:628 +0x6e
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func52({0x4892038, 0xc008b66ff0}, {0x48bf850?, 0xc004dc6040?})
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/processor/abci.go:658 +0x5d
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc000658a80, {0xb8?, 0x3cd6820?}, 0xc004d90d80)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/abci/abci.go:179 +0x622
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc004e21a40?, {0x4892000?, 0x6e5f000?}, 0xc001c004e0?)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x25
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc0005d6a20)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:212 +0x556
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc0005d6a20, 0x6c088e200)
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/nullchain.go:289 +0x1ce
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:132 +0x25
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 26422
	/jenkins/workspace/common/vega-market-sim-fuzzing-develop/extern/vega/core/blockchain/nullchain/server.go:131 +0x22f
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,7 @@
 
 - [11672](https://github.com/vegaprotocol/vega/issues/11672) - Add missing fees in GraphQL bindings.
 - [11681](https://github.com/vegaprotocol/vega/issues/11681) - Account for conflicts inserting funding payment records.
+- [11715](https://github.com/vegaprotocol/vega/issues/11715) - Trade across an `AMMs` base price in two steps since the curves either side will be different.
 - [11684](https://github.com/vegaprotocol/vega/issues/11684) - Better error when `Arbitrum` bridge details are missing from validator configuration.
 - [11696](https://github.com/vegaprotocol/vega/issues/11696) - Add binding for estimate fees API.
 - [11699](https://github.com/vegaprotocol/vega/issues/11699) - Update factors of programs when they are updated.

--- a/core/execution/amm/engine.go
+++ b/core/execution/amm/engine.go
@@ -568,8 +568,22 @@ func (e *Engine) partition(agg *types.Order, inner, outer *num.Uint) ([]*Pool, [
 		// pool is active in range add it to the slice
 		active = append(active, p)
 
+		// we hit a discontinuity where an AMM's two curves meet if we try to trade over its base-price
+		// so we partition the inner/outer price range at the base price so that we instead trade across it
+		// in two steps.
+		boundary := p.upper.low
+		if inner != nil && outer != nil {
+			if boundary.LT(outer) && boundary.GT(inner) {
+				bounds[boundary.String()] = boundary.Clone()
+			}
+		} else if outer == nil && boundary.GT(inner) {
+			bounds[boundary.String()] = boundary.Clone()
+		} else if inner == nil && boundary.LT(outer) {
+			bounds[boundary.String()] = boundary.Clone()
+		}
+
 		// if a pool's upper or lower boundary exists within (inner, outer) then we consider that a sub-level
-		boundary := p.upper.high
+		boundary = p.upper.high
 		if outer == nil || boundary.LT(outer) {
 			bounds[boundary.String()] = boundary.Clone()
 		}

--- a/core/execution/amm/engine_test.go
+++ b/core/execution/amm/engine_test.go
@@ -234,11 +234,14 @@ func testBasicSubmitOrder(t *testing.T) {
 	assert.Equal(t, "2021", ba.String())
 
 	orders = tst.engine.SubmitOrder(agg, num.NewUint(2020), num.NewUint(1990))
-	require.Len(t, orders, 1)
-	assert.Equal(t, "2004", orders[0].Price.String())
-	// note that this volume being bigger than 242367 above means we've moved back to position, then flipped
-	// sign, and took volume from the other curve.
-	assert.Equal(t, 362325, int(orders[0].Size))
+
+	// two orders because we have to split it when we trade across the base-price as thats where we move from one curve to the other.
+	require.Len(t, orders, 2)
+	assert.Equal(t, "2009", orders[0].Price.String())
+	assert.Equal(t, 236855, int(orders[0].Size))
+
+	assert.Equal(t, "1994", orders[1].Price.String())
+	assert.Equal(t, 125470, int(orders[1].Size))
 }
 
 func testSubmitOrderAtBestPrice(t *testing.T) {

--- a/core/integration/features/amm/0090-VAMM-006-014.feature
+++ b/core/integration/features/amm/0090-VAMM-006-014.feature
@@ -348,22 +348,23 @@ Feature: Ensure the vAMM positions follow the market correctly
     # In a single trade, move the mid price to 110
     When the parties place the following orders:
       | party  | market id | side | volume | price | resulting trades | type       | tif     |
-      | party4 | ETH/MAR22 | buy  | 440    | 110   | 1                | TYPE_LIMIT | TIF_GTC |
+      | party4 | ETH/MAR22 | buy  | 440    | 110   | 2                | TYPE_LIMIT | TIF_GTC |
     Then the market data for the market "ETH/MAR22" should be:
       | mark price | trading mode            | ref price | mid price | static mid price | best offer price | best bid price |
       | 94         | TRADING_MODE_CONTINUOUS | 100       | 110       | 110              | 111              | 109            |
     And the following trades should be executed:
       | buyer  | price | size | seller   | is amm |
-      | party4 | 95    | 440  | vamm1-id | true   |
+      | party4 | 94    | 371  | vamm1-id | true   |
+      | party4 | 104   | 69   | vamm1-id | true   |
     # Check the resulting position, vAMM switched from long to short
     When the network moves ahead "1" blocks
 	Then the parties should have the following profit and loss:
       | party    | volume | unrealised pnl | realised pnl | is amm |
-      | party1   | 1      | -5             | 0            |        |
-      | party2   | -1     | 5              | 0            |        |
-      | party3   | -371   | -371           | 0            |        |
-      | party4   | 440    | 0              | 0            |        |
-      | vamm1-id | -69    | 0              | 371          | true   |
+      | party1   | 1      | 4              | 0            |        |
+      | party2   | -1     | -4             | 0            |        |
+      | party3   | -371   | -3710          | 0            |        |
+      | party4   | 440    | 3710           | 0            |        |
+      | vamm1-id | -69    | 0              | 0            | true   |
 
     # Now return the mid price back to 100
     When the parties place the following orders:
@@ -371,7 +372,7 @@ Feature: Ensure the vAMM positions follow the market correctly
       | party5 | ETH/MAR22 | sell | 69     | 100   | 1                | TYPE_LIMIT | TIF_GTC |
     Then the market data for the market "ETH/MAR22" should be:
       | mark price | trading mode            | ref price | mid price | static mid price | best offer price | best bid price |
-      | 95         | TRADING_MODE_CONTINUOUS | 100       | 100       | 100              | 101              | 99             |
+      | 104        | TRADING_MODE_CONTINUOUS | 100       | 100       | 100              | 101              | 99             |
     And the following trades should be executed:
       | buyer    | price | size | seller | is amm |
       | vamm1-id | 104   | 69   | party5 | true   |
@@ -382,9 +383,9 @@ Feature: Ensure the vAMM positions follow the market correctly
       | party1   | 1      | 4              | 0            |        |
       | party2   | -1     | -4             | 0            |        |
       | party3   | -371   | -3710          | 0            |        |
-      | party4   | 440    | 3960           | 0            |        |
+      | party4   | 440    | 3710           | 0            |        |
       | party5   | -69    | 0              | 0            |        |
-      | vamm1-id | 0      | 0              | -250         | true   |
+      | vamm1-id | 0      | 0              | 0            | true   |
 
   @VAMM
   Scenario: 0090-VAMM-013: If other traders trade to move the market mid price to 90 and then move the mid price back to 100 in several trades of varying size, the vAMM will have a position of 0.
@@ -431,7 +432,7 @@ Feature: Ensure the vAMM positions follow the market correctly
       | party5   | 130    | 0              | 0            |        |
       | vamm1-id | 0      | 0              | -257         | true   |
 
-  @VAMM
+  @VAMM3
   Scenario: 0090-VAMM-014: If other traders trade to move the market mid price to 90 and then in one trade move the mid price to 110 then trade to move the mid price to 120 the vAMM will have a larger (more negative) but comparable position to if they had been moved straight from 100 to 120.
     # Move mid price to 90
     When the parties place the following orders:
@@ -455,22 +456,23 @@ Feature: Ensure the vAMM positions follow the market correctly
     # In a single trade, move the mid price to 110
     When the parties place the following orders:
       | party  | market id | side | volume | price | resulting trades | type       | tif     |
-      | party4 | ETH/MAR22 | buy  | 420    | 110   | 1                | TYPE_LIMIT | TIF_GTC |
+      | party4 | ETH/MAR22 | buy  | 420    | 110   | 2                | TYPE_LIMIT | TIF_GTC |
     Then the market data for the market "ETH/MAR22" should be:
       | mark price | trading mode            | ref price | mid price | static mid price | best offer price | best bid price |
       | 95         | TRADING_MODE_CONTINUOUS | 100       | 110       | 110              | 111              | 109            |
     And the following trades should be executed:
       | buyer  | price | size | seller   | is amm |
-      | party4 | 95    | 420  | vamm1-id | true   |
+      | party4 | 94    | 350  | vamm1-id | true   |
+      | party4 | 104   | 70   | vamm1-id | true   |
     # Check the resulting position, vAMM switched from long to short
     When the network moves ahead "1" blocks
 	Then the parties should have the following profit and loss:
       | party    | volume | unrealised pnl | realised pnl | is amm |
-      | party1   | 1      | -5             | 0            |        |
-      | party2   | -1     | 5              | 0            |        |
-      | party3   | -350   | 0              | 0            |        |
-      | party4   | 420    | 0              | 0            |        |
-      | vamm1-id | -70    | 0              | 0            | true   |
+      | party1   | 1      | 4              | 0            |        |
+      | party2   | -1     | -4             | 0            |        |
+      | party3   | -350   | -3150          | 0            |        |
+      | party4   | 420    | 3500           | 0            |        |
+      | vamm1-id | -70    | 0              | -350         | true   |
 
     # Now further increase the mid price, move it up to 120
     When the parties place the following orders:
@@ -478,7 +480,7 @@ Feature: Ensure the vAMM positions follow the market correctly
       | party5 | ETH/MAR22 | buy  | 65     | 120   | 1                | TYPE_LIMIT | TIF_GTC |
     Then the market data for the market "ETH/MAR22" should be:
       | mark price | trading mode            | ref price | mid price | static mid price | best offer price | best bid price |
-      | 95         | TRADING_MODE_CONTINUOUS | 100       | 120       | 120              | 121              | 119            |
+      | 104        | TRADING_MODE_CONTINUOUS | 100       | 120       | 120              | 121              | 119            |
     And the following trades should be executed:
       | buyer  | price | size | seller   | is amm |
       | party5 | 114   | 65   | vamm1-id | true   |
@@ -489,6 +491,6 @@ Feature: Ensure the vAMM positions follow the market correctly
       | party1   | 1      | 14             | 0            |        |
       | party2   | -1     | -14            | 0            |        |
       | party3   | -350   | -6650          | 0            |        |
-      | party4   | 420    | 7980           | 0            |        |
+      | party4   | 420    | 7700           | 0            |        |
       | party5   | 65     | 0              | 0            |        |
-      | vamm1-id | -135   | -1330          | 0            | true   |
+      | vamm1-id | -135   | -700           | -350         | true   |

--- a/core/integration/features/amm/0090-VAMM-028.feature
+++ b/core/integration/features/amm/0090-VAMM-028.feature
@@ -418,7 +418,7 @@ Feature: Ensure the vAMM positions follow the market correctly
     # Now to move from 110 down to 90, the volume ought to be 421 (=347+74)
     When the parties place the following orders:
       | party  | market id | side | volume | price | resulting trades | type       | tif     |
-      | party6 | ETH/MAR23 | sell | 421    | 90    | 1                | TYPE_LIMIT | TIF_GTC |
+      | party6 | ETH/MAR23 | sell | 421    | 90    | 2                | TYPE_LIMIT | TIF_GTC |
     Then the market data for the market "ETH/MAR23" should be:
       | mark price | trading mode            | target stake | supplied stake | open interest | ref price | mid price | static mid price | best offer price | best bid price |
       | 100        | TRADING_MODE_CONTINUOUS | 13915        | 1000           | 348           | 100       | 90        | 90               | 91               | 89             |

--- a/core/integration/features/amm/0090-VAMM-basic.feature
+++ b/core/integration/features/amm/0090-VAMM-basic.feature
@@ -53,8 +53,8 @@ Feature: vAMM rebasing when created or amended
       | trading.terminated | TYPE_BOOLEAN | trading termination |
 
     And the markets:
-      | id        | quote name | asset | liquidity monitoring | risk model            | margin calculator   | auction duration | fees          | price monitoring | data source config     | linear slippage factor | quadratic slippage factor | sla params |
-      | ETH/MAR22 | USD        | USD   | lqm-params           | log-normal-risk-model | margin-calculator-1 | 2                | fees-config-1 | default-none     | termination-oracle     | 1e0                    | 0                         | SLA-22     |
+      | id        | quote name | asset | liquidity monitoring | risk model            | margin calculator   | auction duration | fees          | price monitoring | data source config     | linear slippage factor | quadratic slippage factor | sla params | allowed empty amm levels |
+      | ETH/MAR22 | USD        | USD   | lqm-params           | log-normal-risk-model | margin-calculator-1 | 2                | fees-config-1 | default-none     | termination-oracle     | 1e0                    | 0                         | SLA-22     | 5                        |
 
     # Setting up the accounts and vAMM submission now is part of the background, because we'll be running scenarios 0090-VAMM-006 through 0090-VAMM-014 on this setup
     Given the parties deposit on asset's general account the following amount:
@@ -69,6 +69,7 @@ Feature: vAMM rebasing when created or amended
       | party5 | USD   | 1000000 |
       | vamm1  | USD   | 1000000 |
       | vamm2  | USD   | 1000000 |
+      | vamm3  | USD   | 1000000 |
 
 
     And the parties place the following orders:
@@ -77,8 +78,6 @@ Feature: vAMM rebasing when created or amended
       | party5 | ETH/MAR22 | buy  | 20     | 90    | 0                | TYPE_LIMIT | TIF_GTC | lp1-b     |
       | party1 | ETH/MAR22 | buy  | 1      | 100   | 0                | TYPE_LIMIT | TIF_GTC |           |
       | party2 | ETH/MAR22 | sell | 1      | 100   | 0                | TYPE_LIMIT | TIF_GTC |           |
-      | party3 | ETH/MAR22 | sell | 10     | 110   | 0                | TYPE_LIMIT | TIF_GTC |           |
-      | lp1    | ETH/MAR22 | sell | 10     | 160   | 0                | TYPE_LIMIT | TIF_GTC | lp1-s     |
     When the opening auction period ends for market "ETH/MAR22"
     Then the following trades should be executed:
       | buyer  | price | size | seller |
@@ -212,13 +211,173 @@ Feature: vAMM rebasing when created or amended
 
     When the parties place the following orders:
       | party  | market id | side | volume | price | resulting trades | type       | tif     | reference |
-      | party1 | ETH/MAR22 | buy  | 2      | 200   | 1                | TYPE_LIMIT | TIF_GTC |           |
+      | party1 | ETH/MAR22 | buy  | 2      | 200   | 2                | TYPE_LIMIT | TIF_GTC |           |
 
     Then the parties amend the following AMM:
       | party | market id | amount | slippage | base | lower bound | proposed fee | error                                       |
       | vamm1 | ETH/MAR22 | 100000  | 0.05    | 100  | 95          | 0.03         | cannot remove upper bound when AMM is short |
 
 
+  @VAMM
+  Scenario: Two AMM's incoming order split pro-rata equally
 
-    
+    When the parties submit the following AMM:
+      | party | market id | amount  | slippage | base | lower bound | upper bound | proposed fee |
+      | vamm2 | ETH/MAR22 | 100000  | 0.05     | 100  | 95          | 105         | 0.03         |
+    Then the AMM pool status should be:
+      | party | market id | amount | status        | base | lower bound | upper bound | 
+      | vamm2 | ETH/MAR22 | 100000 | STATUS_ACTIVE | 100  | 95          | 105         | 
+
+    And set the following AMM sub account aliases:
+      | party | market id | alias    |
+      | vamm2 | ETH/MAR22 | vamm2-id |
+
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 99             | 101              | 206             | 184               |
+
+
+    When the parties place the following orders:
+        | party  | market id | side | volume  | price | resulting trades | type       | tif     | reference |
+        | party1 | ETH/MAR22 | sell | 1000    | 10    | 2                | TYPE_LIMIT | TIF_GTC |           |
+    Then the following trades should be executed:
+        | buyer     | price | size | seller    | is amm |
+        | vamm1-id  | 97    | 500  | party1    | true   |
+        | vamm2-id  | 97    | 500  | party1    | true   |
    
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 95             | 96               | 70              | 150               |
+
+  @VAMM
+  Scenario: Two AMM's incoming order split pro-rata unequally
+
+    When the parties submit the following AMM:
+      | party | market id | amount  | slippage | base | lower bound | upper bound | proposed fee |
+      | vamm2 | ETH/MAR22 | 75000   | 0.05     | 100  | 90          | 110         | 0.03         |
+    Then the AMM pool status should be:
+      | party | market id | amount | status        | base | lower bound | upper bound | 
+      | vamm2 | ETH/MAR22 | 75000  | STATUS_ACTIVE | 100  | 90          | 110         | 
+
+    And set the following AMM sub account aliases:
+      | party | market id | alias    |
+      | vamm2 | ETH/MAR22 | vamm2-id |
+
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 99             | 101              | 141             | 125               |
+
+
+    When the parties place the following orders:
+        | party  | market id | side | volume  | price | resulting trades | type       | tif     | reference |
+        | party1 | ETH/MAR22 | sell | 400     | 10    | 2                | TYPE_LIMIT | TIF_GTC |           |
+    Then the following trades should be executed:
+        | buyer     | price | size | seller    | is amm |
+        | vamm1-id  | 98    | 292  | party1    | true   |
+        | vamm2-id  | 98    | 108  | party1    | true   |
+   
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 96             | 98               | 184             | 113               |
+
+
+  @VAMM
+  Scenario: Two AMM's incoming order split pro-rata through base price where one side is much wider
+
+    When the parties submit the following AMM:
+      | party | market id | amount  | slippage | base | lower bound | upper bound | proposed fee |
+      | vamm2 | ETH/MAR22 | 100000  | 0.05     | 100  | 95          | 500         | 0.03         |
+    Then the AMM pool status should be:
+      | party | market id | amount | status        | base | lower bound | upper bound | 
+      | vamm2 | ETH/MAR22 | 100000 | STATUS_ACTIVE | 100  | 95          | 500         | 
+
+    And set the following AMM sub account aliases:
+      | party | market id | alias    |
+      | vamm2 | ETH/MAR22 | vamm2-id |
+
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 99             | 101              | 206             | 92                |
+
+
+    When the parties place the following orders:
+        | party  | market id | side | volume  | price | resulting trades | type       | tif     | reference |
+        | party1 | ETH/MAR22 | sell | 400     | 10    | 2                | TYPE_LIMIT | TIF_GTC |           |
+    Then the following trades should be executed:
+        | buyer     | price | size | seller    | is amm |
+        | vamm1-id  | 99    | 200  | party1    | true   |
+        | vamm2-id  | 99    | 200  | party1    | true   |
+   
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 97             | 99               | 232             | 194               |
+
+
+    # now lets swing back across both AMM's base price, AMM2 will have much less volume on that side 
+    When the parties place the following orders:
+        | party  | market id | side | volume  | price | resulting trades | type       | tif     | reference |
+        | party1 | ETH/MAR22 | buy  | 600     | 115   | 4                | TYPE_LIMIT | TIF_GTC |           |
+    Then the following trades should be executed:
+        | seller     | price | size | buyer    | is amm |
+        | vamm1-id   | 98    | 200  | party1   | true   |
+        | vamm2-id   | 98    | 200  | party1   | true   |
+        | vamm1-id   | 101   | 199  | party1   | true   |
+        | vamm2-id   | 100   | 1    | party1   | true   |
+    
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 102            | 104              | 16              | 163               |
+
+
+
+@VAMM
+  Scenario: 2 AMM's incoming order split pro-rata through, two AMM's are low volume on opposite ends
+
+    When the parties amend the following AMM:
+       | party  | market id | amount  | slippage | base | lower bound | upper bound |   
+       | vamm1  | ETH/MAR22 | 500000  | 0.15     | 2000 | 1995        | 2300        |
+
+
+    When the parties submit the following AMM:
+      | party | market id | amount  | slippage | base | lower bound | upper bound | proposed fee |
+      | vamm2 | ETH/MAR22 | 500000  | 0.05     | 2000 | 1700        | 2005        | 0.03         |
+    Then the AMM pool status should be:
+      | party | market id | amount | status        | base | lower bound | upper bound | 
+      | vamm2 | ETH/MAR22 | 500000 | STATUS_ACTIVE | 2000  | 1700       | 2005        | 
+
+    And set the following AMM sub account aliases:
+      | party | market id | alias    |
+      | vamm2 | ETH/MAR22 | vamm2-id |
+
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 1999           | 2001             | 25              | 23                |
+
+
+    When the parties place the following orders:
+        | party  | market id | side | volume  | price | resulting trades | type       | tif     | reference |
+        | party1 | ETH/MAR22 | sell | 100     | 10    | 2                | TYPE_LIMIT | TIF_GTC |           |
+    Then the following trades should be executed:
+        | buyer     | price  | size | seller    | is amm |
+        | vamm1-id  | 1998   | 99   | party1    | true   |
+        | vamm2-id  | 1998   | 1    | party1    | true   |
+   
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 1995           | 1997             | 30              | 22                |
+
+
+    # now lets swing back across both AMM's base price, AMM2 will have much less volume on that side 
+    When the parties place the following orders:
+        | party  | market id | side | volume  | price | resulting trades | type       | tif     | reference |
+        | party1 | ETH/MAR22 | buy  | 200     | 5000  | 4                | TYPE_LIMIT | TIF_GTC |           |
+    Then the following trades should be executed:
+        | seller    | price | size | buyer     | is amm |
+        | vamm1-id  | 1997  | 99   | party1    | true   |
+        | vamm2-id  | 1998  | 1    | party1    | true   |
+        | vamm1-id  | 2001  | 1    | party1    | true   |
+        | vamm2-id  | 2002  | 99   | party1    | true   |
+   
+    And the market data for the market "ETH/MAR22" should be:
+      | mark price | trading mode             | best bid price | best offer price | best bid volume | best offer volume |
+      | 100        | TRADING_MODE_CONTINUOUS  | 2004           | 2005             | 5               | 19                |

